src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -924,6 +929,37 @@ public Unit getStrategicBombingRaidTarget(final Territory territory, final Colle
     return selected.get();
   }
 
+  @SuppressWarnings("serial")
+  class UnitRenderer extends JLabel implements ListCellRenderer<Unit> {
+
+    public UnitRenderer() {
+      setOpaque(true);
+    }
+
+    @Override
+    public Component getListCellRendererComponent(final JList<? extends Unit> list, final Unit unit, final int index,
+        final boolean isSelected, final boolean cellHasFocus) {
+
+      setText(unit.toString() + ", damage=" + TripleAUnit.get(unit).getUnitDamage());
+      final Optional<ImageIcon> icon = uiContext.getUnitImageFactory().getIcon(unit.getType(), unit.getOwner(),
```

Comment:
very minor, you can write the optional + if statement in one statement:

```uiContext.getUnitImageFactory().getIcon(unit.getType(), unit.getOwner(),
           Matches.unitHasTakenSomeBombingUnitDamage().match(unit), Matches.unitIsDisabled().match(unit)).ifPresent(icon -> setIcon(icon));```

Reply:
Gonna leave it on separate lines as I think its a little easier to read